frontend/packages/console-shared/src/hooks/useUserSettingsCompatibility.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { useUserSettings } from './useUserSettings';
+
+export const useUserSettingsCompatibility = <T>(
+  key: string,
+  storageKey: string,
+  defaultValue?: T,
+): [T | string, React.Dispatch<React.SetStateAction<T>>, boolean] => {
+  const [settings, setSettings, loaded] = useUserSettings<T | string>(
+    key,
+    localStorage.getItem(storageKey) || defaultValue,
+  );
+
+  React.useEffect(
+    () => () => localStorage.removeItem(storageKey),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
```

Comment:
`setSettings` is being called by component using hook or first useEffect on configMap Data fetch (and this will update ConfigMap) so added on unmount else from `useUserSetting` need return patchStatus or something and update based on that

Reply:
@invincibleJai right. Forgot about this call. Make sure to only clear it if we have loaded then:
```suggestion
  React.useEffect(
    () => () => {
      if (loaded) {
        localStorage.removeItem(storageKey);
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [loaded],
  );
```